calibrators/intrinsic_camera_calibrator/intrinsic_camera_calibrator/intrinsic_camera_calibrator/views/image_view.py

```diff
@@ -202,6 +211,162 @@ def set_rendering_alpha(self, value: float):
         """Set the alpha channel of the drawings."""
         self.rendering_alpha = value
 
+    def set_draw_indicators(
+        self,
+        board_speed: float,
+        max_allowed_board_speed: float,
+        skew_percentage: float,
+        board_size_percentage: float,
+        rows_linear_error: float,
+        cols_linear_error: float,
+        pct_err_rows: float,
+        pct_err_cols: float,
+        aspect_ratio: float,
+        pan: float,
+        tilt: float,
+        alpha_indicators: float,
+        value: bool,
+    ):
+        """Set values for indicators."""
+        self.current_board_speed = board_speed
+        self.max_board_allowed_speed = max_allowed_board_speed
+        self.skew_percentage = skew_percentage
+        self.board_size_percentage = board_size_percentage
+        self.rows_linear_error = rows_linear_error
+        self.cols_linear_error = cols_linear_error
+        self.pct_err_rows = pct_err_rows
+        self.pct_err_cols = pct_err_cols
+        self.aspect_ratio = aspect_ratio
+        self.pan = pan
+        self.tilt = tilt
+        self.alpha_indicators = alpha_indicators
+        self.is_draw_indicators = value
+
+    def draw_indicators(self, painter: QPainter, display_size):
+        """Draw indicators for speed, skew, size aspect ratio, angles of the detected board."""
+        color_green = QColor(0.0, 255, 0.0, int(255 * self.alpha_indicators))
+        color_red = QColor(255, 0.0, 0.0, int(255 * self.alpha_indicators))
+        # Change color according tot he speed
+        if self.current_board_speed < self.max_board_allowed_speed:
+            pen = QPen(color_green)
+            brush = QBrush(color_green)
+        else:
+            pen = QPen(color_red)
+            brush = QBrush(color_red)
+        painter.setPen(pen)
+        painter.setBrush(brush)
+        speed_indicator = QRectF(
+            QPointF(0, 0), QSize(display_size.width(), int(display_size.height() * 0.04))
+        )
+        # Draw the rectangle for speed indication
+        painter.drawRect(speed_indicator)
+
+        # Set the font according to the window size
+        font_size = max(10, display_size.height() / 40)
+        font = QFont("Arial", font_size)
+        painter.setFont(font)
+        position_text_speed = QPointF(
+            int(display_size.width() * 0.01), int(display_size.height() * 0.10)
+        )
+        painter.drawText(position_text_speed, "Speed")
+
+        # TODO(Sergio): define percentage to change skew and pct size to change to green
```

Comment:
nit: from now use the github account name

Reply:
f89b3ff4cb6f41ae9f344962700594a3b0d2fee4